airflow/www/views.py

```diff
@@ -714,34 +714,38 @@ def rendered(self):
     @login_required
     @wwwutils.action_logging
     def log(self):
-        BASE_LOG_FOLDER = os.path.expanduser(
-            conf.get('core', 'BASE_LOG_FOLDER'))
+        base_log_folder = os.path.expanduser(conf.get('core', 'BASE_LOG_FOLDER'))
         dag_id = request.args.get('dag_id')
         task_id = request.args.get('task_id')
         execution_date = request.args.get('execution_date')
+        dttm = dateutil.parser.parse(execution_date)
+        iso = dttm.isoformat()
```

Comment:
Refactor this logic  and the logic below with the other places you transform execution date similarly.